napari/layers/image/_image_slice.py

```diff
@@ -0,0 +1,43 @@
+from ...types import ArrayLike
+from ._image_view import ImageView
+
+
+class ImageSlice:
+    """The slice of the image that we are currently viewing.
+
+    Right now this just holds the image and its thumbnail, however future async
+    and multiscale-async changes will likely grow this class a lot.
+
+    Attributes
+    ----------
+    image : ImageView
+        The main image for this slice.
+
+    thumbnail : ImageView
+        The smaller thumbnail image for this slice.
+
+    Example
+    -------
+        # Create with some default image.
+        slice = ImageSlice(default_image)
```

Comment:
this shadows the python built-in `slice` object

Reply:
I was like why didn't it warn me? But it was only in the comment. I changed it to `image_slice`, best to not shadow built-ins in examples.